bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/IRobotHandlerFactory.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.irobot.internal;
+
+import static org.openhab.binding.irobot.IRobotBindingConstants.THING_TYPE_ROOMBA;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.irobot.handler.RoombaHandler;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link IRobotHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author hkuhn42 - Initial contribution
+ * @author Pavel Fedin <pavel_fedin@mail.ru> - rename and update
+ */
+@Component(configurationPid = "binding.irobot", service = ThingHandlerFactory.class)
+@NonNullByDefault
+public class IRobotHandlerFactory extends BaseThingHandlerFactory {
+
+    private final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_ROOMBA);
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> config) {
+        super.activate(componentContext);
+    }
```

Comment:
What's the purpose of this method?

Reply:
I don't remember. :) Probably i was puzzled by the fact that superclass' method has only one argument and subclass' one apparently has two. I have no idea how OSGi figures out what to supply. To me OSGi is a huge magical pandora's box, unbelievably badly documented. Their main page starts with some enterprise blather,  They have a "Developer" page, on attempt to download anything they ask me to fill in my personal data and subscribe to god knows what and why. And simple entering "@activate" in the search box in the header yields "Not found".
So i don't know how to navigate their docs, sorry for that.
Or may be i just don't know what a Java annotation really is and how it maps to some classes / objects / whatever
Anyways, removed.